### PR TITLE
Send ClientID DHCP option in DISCOVER/ACK msgs v2

### PR DESCRIPTION
**Pull request by fnordahl**
This is a continuation of the work in #83 

Please note that it is a work in progress and I need to add tests to actually verify that my hypothesis for a fix holds water.

### CODE REVIEW

File: Documentation/ipvlan.md [hunk 1]
```diff
@@ -38,3 +38,31 @@ Because all ipvlan interfaces share the MAC address with the host interface, DHC
...
+This can be accomplished by setting the "broadcast" flag in the "ipam" section of the network configuration:
+
+```
+{
+	"name": "mynet",
+	"type": "ipvlan",
+	"master": "eth0",
+	"ipam": {
+		"type": "dhcp",
+		"broadcast": true
```
Comment (dcbw) [inline Documentation/ipvlan.md]: Should ipvlan force broadcast to true?  Or should that be left up to the caller of the CNI plugin to handle?

**Comment by dcbw** [comment]
LGTM

**Comment by fnordahl** [comment]
Thank you for your feedback @dcbw 

Currently doing some r&d for how to best create tests for this here: fnordahl/flux-capacitor#1

I will be back with tests suitable for merging with this PR soon.

**Comment by fnordahl** [comment]
@steveeJ I have added unit tests.

I have also added functional tests, but they need a few more adjustments before they work. I have to toss in the towel now. Getting back at it tomorrow.

**Comment by fnordahl** [comment]
I think we need a newer or patched kernel/ipvlan driver to make the functional tests work. Specifically this patch: https://patchwork.ozlabs.org/patch/455225/

Travis-Ci image is currently running Ubuntu 14.04 with the 3.19.0-30 kernel from 15.04 (Vivid). I have the tests working on a newer system here.

Possible strategies going forward:
- [ ] Do we have access to resources elsewhere to run these tests (VM on GCE?)
- [ ] Backport the patch to 3.19.0 and compile the ipvlan module on Travis-Ci before running the tests
- [ ] Make Travis-Ci provide updated images

File: Documentation/ipvlan.md (hunk 1 above)
Comment (fnordahl) [inline Documentation/ipvlan.md]: Good point @dcbw . The current codebase does not seem to have relations to allow this to be deduced automatically, but we sure should aim at making that a reality.

**Comment by dcbw** [comment]
@fnordahl yeah, you'll likely need some of those patches.  3.19 is so, so very old; in fact ipvlan was only added for the 3.19 kernel so it's entirely unsurprising that the driver would have bugs.

**Comment by dcbw** [comment]
@fnordahl but instead of my patches, Mahesh solved the issue a different way.  You'd want:

ba35f8588f474d5bf8988615b04ee722a2684fd2
f631c44bbe1581d18d0aba628b3802d0eacb6373

**Comment by fnordahl** [comment]
> @fnordahl but instead of my patches, Mahesh solved the issue a different way. You'd want:
> 
> ba35f8588f474d5bf8988615b04ee722a2684fd2
> f631c44bbe1581d18d0aba628b3802d0eacb6373

Thanks @dcbw ! f631c44bbe1581d18d0aba628b3802d0eacb6373 applies clean on the 3.19 sources and it suffices for the tests to work.

Patch updated, and now with consistently successful functional tests!

**Comment by squeed** [comment]
Hm, it looks like this took a year-long nap. What is needed to bring it back to life?

**Comment by fnordahl** [comment]
@squeed feedback. Previous feedback was missing tests, functional tests has since been added but the PR has been ignored nonetheless. Code is probably still good but needs a re-base.

Should I interpret your query as a sign of interest in having this merged?

**Comment by squeed** [comment]
Indeed, I think a more well-rounded DHCP plugin would be a very good idea. The lack of DNS is also frustrating.

However, the repository has split, so please consider re-submitting for the Plugins repo.

**Pull request closed without merging**